opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -123,6 +123,10 @@ message Span {
   // [Optional].
   //
   // See https://www.w3.org/TR/trace-context-2/#trace-flags for the flag definitions.
+  //
+  // Bits 8 and 9 are reserved for representing the 3 states of whether a span's parent
+  // is remote. The states are unknown, is not remote, is remote.
```

Comment:
since this is a bit field, shouldn't we use 1 and 3 or 2 and 3, so that one bit represents "it is known whether remote or not" and the other represents the remoteness?

Reply:
@Oberon00 sure, that works.

Reply:
I've included an enum using bit 8 as unset/set and bit 9 as the remoteness.

Reply:
After thinking about it, I think the flags field is currently underdefined or maybe even defined misleadingly: https://github.com/open-telemetry/opentelemetry-proto/pull/503

Depending on how it's meant to be defined, and also considering usage on Links, it might be confusing to use "parent" in the name of the flag, i.e. instead of SPAN_FLAGS_PARENT_IS_REMOTE we might need to call it SPAN_FLAGS_IS_REMOTE, if the flags generally come from the parent (since it's also SPAN_FLAGS_TRACE_FLAGS_MASK, and not SPAN_FLAGS_PARENT_TRACE_FLAGS_MASK)

Reply:
>it might be confusing to use "parent" in the name of the flag

This is about recording the fact in the _current_ span that its _parent_ span is _remote_ (the definition of "remote" AFAIK is that it describes an operation in a different service). I think using the word "parent" is important. It is the parent that is remote, not the current span.

Reply:
But the same is the case e.g. for the W3C sampled flag ... or is it not?

Reply:
@yurishkuro That does not really make sense, and also contradicts the description of the PR where the flag was added, see https://github.com/open-telemetry/opentelemetry-proto/issues/507

Reply:
@Oberon00 see my comment in https://github.com/open-telemetry/opentelemetry-proto/issues/507#issuecomment-1727202313

Reply:
> But the same is the case e.g. for the W3C sampled flag ... or is it not?

@Oberon00 Do you agree that the discussion in #507 (mainly around the W3C trace flags) is not directly related to this PR? Here the two newly introduced bits really refer to describing the current span from which we want to know whether the `PARENT` is remote. So it MUST be `SPAN_FLAGS_PARENT_IS_REMOTE` (not SPAN_FLAGS_IS_REMOTE). Otherwise it would defeat the initial intent of what we are encoding into these two flags. 

Can we resolve this here?

Reply:
I disagree that this is unrelated to #507, but I think there is now enough agreement on #507 about the semantics of this field that we can conclude that PARENT should be used in the name.